packages/hash/api/src/model/knowledge/block.model.ts

```diff
@@ -140,31 +140,31 @@ export default class extends EntityModel {
   ): Promise<void> {
     const { newBlockDataEntity, actorId } = params;
     const outgoingBlockDataLinks = await this.getOutgoingLinks(graphApi, {
-      linkTypeModel: SYSTEM_TYPES.linkType.blockData,
+      linkEntityTypeModel: SYSTEM_TYPES.linkEntityType.blockData,
     });
 
     const outgoingBlockDataLink = outgoingBlockDataLinks[0];
 
     if (!outgoingBlockDataLink) {
       throw new Error(
-        `Block with entityId ${this.entityId} does not have an outgoing block data link`,
+        `Block with entityId ${this.baseId} does not have an outgoing block data link`,
       );
     }
 
     if (
-      outgoingBlockDataLink.targetEntityModel.entityId ===
-      newBlockDataEntity.entityId
+      outgoingBlockDataLink.rightEntityModel.entityUuid ===
+      newBlockDataEntity.entityUuid
```

Comment:
Should we be comparing `baseId` here?

Reply:
We could, but what's the benefit? Wouldn't it be the same result, as we're looking for watching entities here? Not whether they're owned by the same account?

Reply:
Well the general philosophy is that an entity is _not_ uniquely defined by it's UUID, it's uniquely defined by its `entityId`. So we should be comparing using that.

Reply:
Okay, have changed this in https://github.com/hashintel/hash/pull/1435/commits/3cb6ed29868d750d40108e418521ade0b2bf056f. I'm about unsure about the implications of this once we support moving entities from one account to another, but we can address this then.